core/trino-spi/src/main/java/io/trino/spi/type/BigintType.java

```diff
@@ -56,4 +56,10 @@ public int hashCode()
     {
         return Optional.of(new Range(Long.MIN_VALUE, Long.MAX_VALUE));
     }
+
+    @Override
+    public Optional<?> getNearestGreaterValue(Object value)
```

Comment:
Could we have a different API like `Stream<?> getRangeIterator(Range range)` instead ?

Reply:
Good suggestion, thanks!
Please take a look at: https://github.com/trinodb/trino/pull/9868/commits/9da363da19f4588fdaf0ef416fc9e5d4ed3408fe